plugins/modules/ec2_asg_lifecycle_hook.py

```diff
@@ -150,23 +154,30 @@ def create_lifecycle_hook(connection, module):
         module.fail_json_aws(e, msg="Failed to get Lifecycle Hook")
 
     if not existing_hook:
-        changed = True
-    else:
-        # GlobalTimeout is not configurable, but exists in response.
-        # Removing it helps to compare both dicts in order to understand
-        # what changes were done.
-        del(existing_hook[0]['GlobalTimeout'])
-        added, removed, modified, same = dict_compare(lch_params, existing_hook[0])
-        if added or removed or modified:
-            changed = True
-
-    if changed:
         try:
+            return_object['changed'] = True
             connection.put_lifecycle_hook(**lch_params)
+            return_object['lifecycle_hook_info'] = connection.describe_lifecycle_hooks(
+                AutoScalingGroupName=asg_name, LifecycleHookNames=[lch_name])['LifecycleHooks']
         except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
             module.fail_json_aws(e, msg="Failed to create LifecycleHook")
 
-    return(changed)
+    else:
+        added, removed, modified, same = dict_compare(lch_params, existing_hook[0])
+        if modified:
+            # GlobalTimeout is not configurable, but exists in response.
+            # Removing it helps to compare both dicts in order to understand
+            # what changes were done.
+            del(existing_hook[0]['GlobalTimeout'])
```

Comment:
I get the logic here for only reporting changed on `modified`, but in that case it looks like you can get rid of the `del` (and the comments with it), since it's not really being used

Reply:
@jatorcasso yes,that makes sense, as I'm just trying to add integration tests here, didn't wanted to make any major changes to the module unless required. 
I can go ahead and remove it if @alinabuzachis or @jillr thinks that's a good idea.

Reply:
If the code is not needed we should remove it so that it has less chance of introducing a big later.  This small of a code change is fine I think.